pkg/migrations/00020_add_latest_envelopes.up.sql

```diff
@@ -0,0 +1,37 @@
+CREATE TABLE gateway_envelopes_latest (
+    originator_node_id int PRIMARY KEY,
+    originator_sequence_id bigint NOT NULL,
+    gateway_time timestamp NOT NULL
+);
+
+CREATE OR REPLACE FUNCTION update_latest_envelope() RETURNS trigger AS $$
+BEGIN
+    INSERT INTO gateway_envelopes_latest
+    VALUES (NEW.originator_node_id, NEW.originator_sequence_id, NEW.gateway_time)
+    ON CONFLICT (originator_node_id)
+        DO UPDATE
+        SET originator_sequence_id = EXCLUDED.originator_sequence_id,
+            gateway_time = EXCLUDED.gateway_time;
+    RETURN NULL;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TRIGGER gateway_latest_upd
+    AFTER INSERT ON gateway_envelopes
+    FOR EACH ROW EXECUTE FUNCTION update_latest_envelope();
```

Comment:
Doing this for every row seems kind of heavy. Is there a way to do this once for each transaction from the app?